src/raftstore/store/snap.rs

```diff
@@ -61,223 +117,517 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileBuilder {
+    fn build(&mut self, snap: &DbSnapshot, region: &Region) -> raft::Result<()>;
```

Comment:
It's not working like that, the `Builder` is used in region worker, which doesn't pass `SendSnapshotFile` to server.
I guess there is a better name for this class, which create a snapshot file from a rocksdb snapshot. But it doesn't come to my head at this moment.